src/olympia/addons/decorators.py

```diff
@@ -55,3 +58,15 @@ def addon_view_factory(qs):
     # GOOD: lambda: Addon.objects.valid().filter(type=1)
     # BAD: Addon.objects.valid()
     return functools.partial(addon_view, qs=qs)
+
+
+def require_submissions_enabled(f):
+    """Require the enable-submissions waffle flag to be enabled."""
+
+    @functools.wraps(f)
+    def wrapper(request, *args, **kw):
+        if waffle.flag_is_active(request, 'enable-submissions'):
+            return f(request, *args, **kw)
+        return submissions_disabled_response()
```

Comment:
Sorry, one more change with the logic: we could have a single decorator actually. Inside, you can return a `Response` (API) or a `TemplateResponse` (non-API) by doing a check on `request.is_api`. And I don't think we need to have the code in utils, it's short enough and never re-used anywhere else, we could have it directly in the decorator IMHO. That allows you to share the bit that retrieves the note between the 2 code paths as well.